frontend/src/app/modules/leader/leader-board/leader-board.component.ts

```diff
@@ -56,19 +90,131 @@ export class LeaderBoardComponent extends BaseComponent implements OnInit {
             .pipe(takeUntil(this.unsubscribe$))
             .subscribe({
                 next: (users) => {
+                    this.handleUsersUpdate(users);
+                },
+                error: () => {
                     this.loading = false;
-                    if (!users.length) {
-                        this.isLastPage = true;
+                    this.toastrNotification.showError('Server connection error');
+                },
+            });
+    }
 
-                        return;
-                    }
-                    this.users = [...this.users, ...users];
-                    this.usersToShow = this.users;
+    private getFriends() {
+        if (this.isLastFriendsPage) {
+            return;
+        }
+
+        this.friendsPage.pageNumber++;
+        this.loading = true;
+
+        this.userService
+            .getFriendsLeaderBoard(this.friendsPage)
+            .pipe(takeUntil(this.unsubscribe$))
+            .subscribe({
+                next: (users) => {
+                    this.handleFriendsUpdate(users);
                 },
                 error: () => {
                     this.loading = false;
                     this.toastrNotification.showError('Server connection error');
                 },
             });
     }
+
+    private getCurrentUser() {
+        this.userService
+            .getCurrentUser()
+            .pipe(takeUntil(this.unsubscribe$))
+            .subscribe({
+                next: (user) => {
+                    this.currentUser = user;
+                    this.userFriendsIds = user.friendships.map((f) => f.friendId);
+                },
+                error: () => {
+                    this.toastrNotification.showError('Server connection error');
+                },
+            });
+    }
+
+    public isCurrentUser(user: IUser): boolean {
+        return user && this.currentUser && user.id === this.currentUser.id;
+    }
+
+    public isAddFriendButtonVisible(user: IUser): boolean {
+        return !this.isCurrentUser(user) && !this.isFriend(user);
+    }
+
+    public isPendingFriendship(user: IUser): boolean {
+        return !this.isCurrentUser(user) && this.getFriendshipStatus(user) === FriendshipStatus.Pending;
+    }
+
+    public isAcceptedFriendship(user: IUser): boolean {
+        return !this.isCurrentUser(user) && this.getFriendshipStatus(user) === FriendshipStatus.Accepted;
+    }
```

Comment:
can be combined, the different is only FriendshipStatus

Reply:
if you meant this 
```
public isFriendshipStatus(user: IUser, status: FriendshipStatus): boolean {
    return !this.isCurrentUser(user) && this.getFriendshipStatus(user) === status;
}
```
```
<div *ngIf="isFriendshipStatus(user, FriendshipStatus.Pending)" class="dot-flashing"></div>
<button class="user-icon-btn" *ngIf="isFriendshipStatus(user, FriendshipStatus.Accepted)">
        <fa-icon class="user-icon" [icon]="['fas', 'user-group']"></fa-icon>
</button>
```
I get this error, and FriendshipStatus import is present in the ts file
![image](https://github.com/BinaryStudioAcademy/bsa-2023-leetwars/assets/56970649/86af92ff-c76a-4c5e-abfa-4a5f01e37209)

Reply:
look how we did it in other places, or use string value in html